tests/integration_tests/performance/test_net_tput.py

```diff
@@ -0,0 +1,330 @@
+        """Build the iperf command applying given params."""
+        self.ssh_connection = net_tools.SSHConnection(self.microvm.ssh_config)
+
+        cmd = '{} -c {} -t{} -f MBytes -J -b {}M {} {} -A {} -P {}'.format(
+                IPERF,
+                params['host_ip'],
+                ITERATION_DURATION,
+                params['bw'],
+                params['mode'],
+                params['protocol'],
+                PINNING_CPU,
+                params['iperfs'],
+            )
+
+        return cmd
+
+    def spawn_iperf_guest(self, iperf_cmd):
+        """Start iperf in client mode and return resulting JSON."""
+        _, stdout, stderr = self.ssh_connection.execute_command(iperf_cmd)
+        assert stderr.read() == ''
+
+        out = stdout.read()
+        return json.loads(out)
+
+
+class IperfUdpClient(IperfClient):
+    def __init__(self, host_ip, iperfs, microvm):
+        super().__init__('udp', host_ip, iperfs, microvm)
+
+    def create_statistics(self):
+        """Create a map which holds metric statistics registered for UDP."""
+        return {
+            'tput': Statistics(),
+            'lost': Statistics(),
+            'pps': Statistics(),
+            'jitter': Statistics(),
+        }
+
+    def build_iperf_cmd(self, params):
+        """Build the iperf command applying given params."""
+        return super().build_iperf_cmd(params)
+
+    def parse_result(self, result, params, udp_statistics):
+        """Parse iperf UDP test output and return result JSON."""
+        test = 'udp-{}-bw{}'.format(
+                    'h2g' if params['mode'] == '-R' else 'g2h', 
+                    params['bw'], 
+                )
+
+        totals = result['end']['sum']
+        total_bytes = int(totals['bytes'])
+        total_packets = int(totals['packets'])
+        lost_packets = int(totals['lost_packets'])
+        duration = float(totals['seconds'])
+        jitter = round(float(totals['jitter_ms']), 2)
+        tput = round((total_bytes*8) / (1024*1024*duration), 2)
+        pps = round(total_packets / duration, 2)
+
+        if test not in udp_statistics:
+            udp_statistics[test] = self.create_statistics()
+
+        udp_statistics[test]['tput'].add(tput)
+        udp_statistics[test]['lost'].add(lost_packets)
+        udp_statistics[test]['pps'].add(pps)
+        udp_statistics[test]['jitter'].add(jitter)
+
+        return {
+            'test': test,
+            'tput': tput,
+            'lost': lost_packets,
+            'pps': pps,
+            'jitter': jitter,
+        }
+
+    def run_iperf_client(self, results):
+        """Build udp parameters upon the common ones."""
+        udp_statistics = {}
+
+        for _ in range(RUNS):
+            for params in super().run_iperf_client():
+                cmd = self.build_iperf_cmd(params)
+                iperf_out = super().spawn_iperf_guest(cmd)
+                result = self.parse_result(iperf_out, params, udp_statistics)
```

Comment:
These 3 lines are also almost completely duplicated for `IperfTcpClient` and `IperfUdpClient`. Can we have this logic inside `IperfClient` which is common for both ?